src/components/core/base/VisualizationLoadingHOC.tsx

```diff
@@ -79,22 +104,28 @@ export function visualizationLoadingHOC<T extends ICommonVisualizationProps & ID
     autoExecuteDataSource: boolean = true
 ): React.ComponentClass<T> {
     class LoadingHOCWrapped
-        extends React.Component<T & ILoadingInjectedProps, IVisualizationLoadingState> {
+        extends React.Component<T & ILoadingInjectedProps & IVisualizationLoadingProps, IVisualizationLoadingState> {
 
         public static defaultProps: Partial<T & ILoadingInjectedProps> = InnerComponent.defaultProps;
 
         protected subject: ISubject<IExecutionDataPromise>;
         protected pagePromises: IExecutionDataPromise[];
         protected hasUnmounted: boolean;
 
-        constructor(props: T & ILoadingInjectedProps) {
+        private sdk: SDK;
+
+        constructor(props: T & ILoadingInjectedProps & IVisualizationLoadingProps) {
             super(props);
 
             this.state = {
                 isLoading: false,
                 result: null,
                 error: null
             };
+
+            this.sdk = (props.sdk || createSdk()).clone();
```

Comment:
why do you call clone on freshly created sdk?
is not this better?
this.sdk = props.sdk ? props.sdk.clone() : createSdk();